src/app/app.test.tsx

```diff
@@ -0,0 +1,51 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { fireEvent, render, RenderAPI, waitFor } from '@testing-library/react-native';
+import { TestRootComponent } from '@tests/helpers';
+import React from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+import { App } from './app';
+
+describe('App', () => {
+  let component: RenderAPI;
+  let tabBarItems: Array<ReactTestInstance>;
+
+  function initComponent(): RenderAPI {
+    return render(
+      <TestRootComponent>
+        <App />
+      </TestRootComponent>
+    );
+  }
+
+  beforeAll(() => {
+    jest.spyOn(AsyncStorage, 'getItem').mockImplementation((key) => {
+      if (key === 'token') {
+        return Promise.resolve('some-demo-token');
+      }
+    });
+  });
+
+  beforeEach(async () => {
+    component = initComponent();
+
+    await waitFor(() => {
+      tabBarItems = component.getAllByTestId('tab-bar-item');
+    });
+  });
+
+  it('should render two tab bar items', async () => {
+    expect(tabBarItems).toHaveLength(2);
+  });
+
+  it('should navigate to Profile and Home screens by press on the tab bar items', async () => {
```

Comment:
`AppTabBar` can't be used without `Tab.Navigator` in [MainNavigation](https://github.com/RonasIT/react-native-starter/blob/main/src/app/main/navigation.tsx). `Tab.Navigator` passes necessary parameters to `AppTabBar` and it would be more difficult to make their mocks

Also `App` is the top of the navigation hierarchy so we decided to write these tests here